### PR TITLE
fix build failure for go1.9

### PR DESCRIPTION
**Pull request by balamurugana**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by harshavardhana — APPROVED** [review]
Can you rebase your changes. LGTM

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Looks like travis has failed @balamurugana can you take a look? 

```
--- FAIL: TestPosixReadFile (0.00s)
	posix_test.go:1083: Case: 11 struct { volume string; fileName string; offset int64; bufSize int; expectedBuf []uint8; expectedErr error }{volume:"success-vol", fileName:"myobject", offset:-1, bufSize:5, expectedBuf:[]uint8(nil), expectedErr:(*os.PathError)(0xc4316289c0)}, expected: readat /tmp/minio-026818568/success-vol/myobject: negative offset, got: read /tmp/minio-026818568/success-vol/myobject: invalid argument
```

File: .travis.yml [hunk 1]
```diff
@@ -27,4 +27,4 @@ after_success:
 - bash <(curl -s https://codecov.io/bash)
 
 go:
-- 1.8.3
+- 1.9
```
Comment (harshavardhana) [inline .travis.yml]: For the time being we can enable both @balamurugana ? i.e keep both `1.8.3` and `1.9`

**Review by harshavardhana — CHANGES_REQUESTED** [review]
We can discuss about taking this PR in with @abperiasamy since we don't need to be quickly migrating to go1.9 right away.  And these tests will fail on our local setups.

File: cmd/posix_test.go [hunk 2]
```diff
@@ -1021,20 +1022,11 @@ func TestPosixReadFile(t *testing.T) {
...
-						Err:  syscall.Errno(0x57), // invalid parameter
-					}
-				}
-				return &os.PathError{
-					Op:   "read",
-					Path: slashpath.Join(path, "success-vol", "myobject"),
-					Err:  os.ErrInvalid,
-				}
-			}(),
+			&os.PathError{
```
Comment (vadmeste) [inline cmd/posix_test.go]: Can you explain a little why this change is needed ? it seems (from travis error log) that windows still returns "invalid parameter", is Linux behavior changed in 1.9 and returning "negative offset" instead ?

Comment (balamurugana) [reply cmd/posix_test.go]: Appveyor hasn't moved to go1.9 yet (even I specified in the configuration), hence the failure

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Can you rebase this PR @balamurugana we can finalize on moving to go1.9.1

**Comment by balamurugana** [comment]
@harshavardhana its already moved.   Do you see any missing thing?

**Review by vadmeste — APPROVED** [review]
LGTM

**Review by harshavardhana — APPROVED** [review]

**Comment by deekoder** [comment]
I will merge after codecov finishes.
